app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java

```diff
@@ -82,6 +87,16 @@ public OpenFoodAPIClient(Activity activity) {
     }
 
     private OpenFoodAPIClient(String apiUrl) {
+        if (Build.VERSION.SDK_INT == 24) {
+            OkHttpClient.Builder builder = new OkHttpClient.Builder();
+            ConnectionSpec spec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+                    .tlsVersions(TlsVersion.TLS_1_2)
+                    .cipherSuites(CipherSuite.TLS_DHE_RSA_WITH_AES_128_GCM_SHA256)
+                    .build();
+            builder.connectionSpecs(Collections.singletonList(spec));
+
+            httpClient = builder.build();
```

Comment:
Hi @sachincool; I'm afraid this would overwrite the timeout settings. There are more OkHttpClient instantations in the code (`AppModule.java` and `OpenFoodAPIServiceTest.java` should also take advantage of this fix). Without doing so other features fail (for example: trying to get the categories fails with a TLS handshake error)

Not sure if it's the best approach, but we could define a helper function to provide a base OkHttpClient.Builder object. Just an idea:

```
public static OkHttpClient.Builder HttpClientBuilder() {
    OkHttpClient.Builder builder = new OkHttpClient.Builder()
            .connectTimeout(5000, TimeUnit.MILLISECONDS)
            .readTimeout(30000, TimeUnit.MILLISECONDS)
            .writeTimeout(30000, TimeUnit.MILLISECONDS);

    if (Build.VERSION.SDK_INT == 24) {
        ConnectionSpec spec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
                .tlsVersions(TlsVersion.TLS_1_2)
                .cipherSuites(CipherSuite.TLS_DHE_RSA_WITH_AES_128_GCM_SHA256)
                .build();
        builder.connectionSpecs(Collections.singletonList(spec));
    }

    return builder;
}
```
Does it make sense?

Reply:
Okay i found a  perfect Solution @policorn @itchix  This works great. Tested on 7.0 aswell as 5.1 
the 3 certs are an Upgrade  since the 7.0  needs last Cert for an successful Request. But other versions can use the other 2 Certs for security purposes. 

```
 static ConnectionSpec spec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
            .tlsVersions(TlsVersion.TLS_1_2)
            .cipherSuites(
                    CipherSuite.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
                    CipherSuite.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
                    CipherSuite.TLS_DHE_RSA_WITH_AES_128_GCM_SHA256)
            .build();

    private static OkHttpClient httpClient = new OkHttpClient.Builder()
            .connectTimeout(5000, TimeUnit.MILLISECONDS)
            .readTimeout(30000, TimeUnit.MILLISECONDS)
            .writeTimeout(30000, TimeUnit.MILLISECONDS)
            .connectionSpecs(Collections.singletonList(spec))
            .build();
```